src/streaming/controllers/ListMpdController.js

```diff
@@ -71,9 +73,28 @@ function ListMpdController() {
         });
     }
 
-    function _onLinkedPeriodsLoaded({ manifest, linkedPeriods}) {
+    function _onLinkedPeriodsLoaded({ manifest, linkedPeriods }) {
         currentManifest = manifest;
         linkedPeriodList = linkedPeriods;
+
+        if (manifest.Period[0].start && manifest.Period[0].start !== 0) {
+            throw new Error('The first period in a list MPD must have start time equal to 0');
+        } else {
+            manifest.Period[0].start = 0;
+        }
```

Comment:
You can simplify the if expression just to manifest.Period[0].start. Because if `start` is equal to 0 the full expression is false.

Also the else is not necessary because you are doing a throw Error.
```suggestion
        if (manifest.Period[0].start) {
            throw new Error('The first period in a list MPD must have start time equal to 0');
        } 
        
        manifest.Period[0].start = 0;
```